easybuild/tools/utilities.py

```diff
@@ -300,22 +300,21 @@ def time2str(delta):
     if not isinstance(delta, datetime.timedelta):
         raise EasyBuildError("Incorrect value type provided to time2str, should be datetime.timedelta: %s", type(delta))
 
-    delta_secs = delta.days * 3600 * 24 + delta.seconds + delta.microseconds / 10**6
-
-    if delta_secs < 60:
-        res = '%d sec' % int(delta_secs)
-    elif delta_secs < 3600:
-        mins = int(delta_secs / 60)
-        secs = int(delta_secs - (mins * 60))
-        res = '%d min %d sec' % (mins, secs)
-    else:
-        hours = int(delta_secs / 3600)
-        mins = int((delta_secs - hours * 3600) / 60)
-        secs = int(delta_secs - (hours * 3600) - (mins * 60))
-        hours_str = 'hours' if hours > 1 else 'hour'
-        res = '%d %s %d min %d sec' % (hours, hours_str, mins, secs)
+    delta_secs = delta.total_seconds()
 
-    return res
+    hours = int(delta_secs / 3600)
+    delta_secs -= hours * 3600
+    mins = int(delta_secs / 60)
+    secs = int(delta_secs - mins * 60)
```

Comment:
can be even more concise:
```suggestion
    hours, remainder = divmod(s, 3600)
    mins, secs = divmod(remainder, 60)

```

Reply:
I like it, thanks! And no, the results of divmod are already ints :)

Reply:
oh it depends on the type of delta_secs. if it's float, the result is also float ;)

Reply:
Oh, yes. It's a float here... :/
Then you are right.